lib/srv/desktop/windows_server.go

```diff
@@ -294,14 +298,23 @@ func NewWindowsService(cfg WindowsServiceConfig) (*WindowsService, error) {
 		return nil, trace.Wrap(err, "fetching cluster name")
 	}
 
-	// Here we assume the LDAP server is an Active Directory Domain Controller,
-	// which means it should also be a DNS server that can resolve Windows hosts.
-	dnsServer, _, err := net.SplitHostPort(cfg.LDAPConfig.Addr)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	dialer := &net.Dialer{Timeout: dnsDialTimeout}
```

Comment:
I would make `dnsResolver` optional on `WindowsService` since you only need it for AD.

That means you can move this dialer inside the if statement, and you can update `lookupDesktop()` to only do the 2nd DNS query if `dnsResolver` is not nil.

Reply:
Yes, that makes sense, I changed it